File: app/src/main/java/com/mytlogos/enterprise/model/MediumType.java
package com.mytlogos.enterprise.model;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class MediumType {
    public static final int ALL = MediumType.ANIME | MediumType.MANGA | MediumType.NOVEL | MediumType.SERIES;
    public static final int NOVEL = 0x1;
    public static final int MANGA = 0x2;
    public static final int ANIME = 0x4;
    public static final int SERIES = 0x8;
    public static final int TEXT = 0x1;
    public static final int AUDIO = 0x2;
    public static final int VIDEO = 0x4;
    public static final int IMAGE = 0x8;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef(value = {TEXT, IMAGE, AUDIO, VIDEO})
    public @interface Medium {

    }

    public static int addMediumType(int mediumType, int toAdd) {
        return mediumType | toAdd;
    }

    public static int removeMediumType(int mediumType, int toRemove) {
        return mediumType & ~toRemove;
    }

    public static int toggleMediumType(int mediumType, int toToggle) {
        return mediumType ^ toToggle;
    }

    public static boolean is(int type, int toCheck) {
        return (type & toCheck) == toCheck;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/model/MediumType.kt
package com.mytlogos.enterprise.model

import androidx.annotation.IntDef

const val NOVEL = 0x1
const val MANGA = 0x2
const val ANIME = 0x4
const val SERIES = 0x8
const val ALL = ANIME or MANGA or NOVEL or SERIES
const val TEXT = 0x1
const val AUDIO = 0x2
const val VIDEO = 0x4
const val IMAGE = 0x8

fun addMediumType(mediumType: Int, toAdd: Int): Int {
    return mediumType or toAdd
}

fun removeMediumType(mediumType: Int, toRemove: Int): Int {
    return mediumType and toRemove.inv()
}

fun toggleMediumType(mediumType: Int, toToggle: Int): Int {
    return mediumType xor toToggle
}

fun isType(type: Int, toCheck: Int): Boolean {
    return type and toCheck == toCheck
}

@Retention(AnnotationRetention.SOURCE)
@IntDef(value = [TEXT, IMAGE, AUDIO, VIDEO])
annotation class MediumType